### ISSUE TITLE
Bazel doesn't respect XDG_CACHE_HOME

### ISSUE THREAD

**Issue by ebyrne242** [bazelbuild/bazel#16937]
### Description of the bug:

Bazel writes caches, often many megabytes in size to $USER/.cache/bazel on Linux.  It's common for users on NFS home dirs to set the XDG_CACHE_HOME environment variable to put their caches on a local disk instead.  However, bazel ignores this variable and continues to write to ~/.cache/bazel.

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

export XDG_CACHE_HOME=/tmp/$USER/bazel

build anything with bazel

bazel still writes output to ~/.cache/bazel

### Which operating system are you running Bazel on?

CentOS 7.9

### If `bazel info release` returns `development version` or `(@non-git)`, tell us how you built Bazel.

```
$ bazel info release

WARNING: Output base '/data/homes/nfs/home/user/.cache/bazel/_bazel_user/33181acb49c359f0b2bfbb3335eecb1f' is on NFS. This may lead to surprising failures and undetermined behavior.

release 5.3.2

```

### Have you found anything relevant by searching the web?

Suggestions that TEST_TMPDIR is the only way to influence the output dir, along with suggestions it should not be used.

**Comment by sketch242** [comment]
It's not code specific, you can use anything.
```$ git clone https://github.com/bazelbuild/examples
Cloning into 'examples'...
remote: Enumerating objects: 1995, done.
remote: Counting objects: 100% (600/600), done.
remote: Compressing objects: 100% (410/410), done.
remote: Total 1995 (delta 219), reused 441 (delta 147), pack-reused 1395
Receiving objects: 100% (1995/1995), 542.98 KiB | 7.87 MiB/s, done.
Resolving deltas: 100% (763/763), done.
$ cd examples/cpp-tutorial/stage1/
$ export $XDG_CACHE_HOME=/data/cache/user
$  readlink -m $HOME
/data/homes/nfs/home/user
$ rm -rf ~/.cache/bazel /data/cache/user/bazel
$ bazel build //main:hello-world
Extracting Bazel installation...
Starting local Bazel server and connecting to it...
INFO: Analyzed target //main:hello-world (37 packages loaded, 163 targets configured).
INFO: Found 1 target...
Target //main:hello-world up-to-date:
  bazel-bin/main/hello-world
INFO: Elapsed time: 6.790s, Critical Path: 0.35s
INFO: 6 processes: 4 internal, 2 processwrapper-sandbox.
INFO: Build completed successfully, 6 total actions
$ ls -ld/data/cache/user/bazel/
ls: cannot access /data/cache/user/bazel/: No such file or directory
$ ls -ld /data/homes/nfs/home/user/.cache/bazel/_bazel_user/8fb73c6900510c4b2c71adc08d4544df
drwxr-xr-x. 6 user users 4096 Dec  7 13:10 /data/homes/nfs/home/user/.cache/bazel/_bazel_user/8fb73c6900510c4b2c71adc08d4544df
```

**Comment by tetromino** [comment]
We should fix this.

**Comment by tetromino** [comment]
... and until this is fixed, as a workaround, you can do `bazel --output_user_root=/tmp/$USER/bazel build ...` - see https://bazel.build/remote/output-directories#layout

**Comment by Wyverald** [comment]
@bazel-io fork 7.2.0

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.2.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/7.2.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=7.2.0rc1. Thanks!
